### DISCUSSION TITLE
Networking & Kubernetes

### DISCUSSION THREAD

**Discussion by TobiasDeBruijn** [haugene/docker-transmission-openvpn#1898] [Q&A] [answered]
Hi there,

I've been migrating my old K8s cluster to my new K8s cluster recently. Of course, not leaving Transmission behind. After setting up the whole manifest, the web UI is reachable, however Transmission is not reachable from other pods in the same namespace. I've verified all Pods can talk to all other Pods, with the exception of Transmission:
```
/ # curl -v -S http://lidarr.homelab.svc.cluster.local:8686
*   Trying 10.80.7.187:8686...
* TCP_NODELAY set
* Connected to lidarr.homelab.svc.cluster.local (10.80.7.187) port 8686 (#0)
> GET / HTTP/1.1
> Host: lidarr.homelab.svc.cluster.local:8686
> User-Agent: curl/7.67.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 303 See Other
< Date: Sat, 07 Aug 2021 23:42:41 GMT
< Content-Type: text/html
< Server: Kestrel
< Transfer-Encoding: chunked
< Location: /login?returnUrl=/
<
* Connection #0 to host lidarr.homelab.svc.cluster.local left intact

# curl -v -S http://transmission.homelab.svc.cluster.local:9091
*   Trying 10.80.4.37:9091...
* TCP_NODELAY set
* connect to 10.80.4.37 port 9091 failed: Operation timed out
* Failed to connect to transmission.homelab.svc.cluster.local port 9091: Operation timed out
* Closing connection 0
curl: (28) Failed to connect to transmission.homelab.svc.cluster.local port 9091: Operation timed out
/ #
```
The Transmission service is called `transmission`, all objects live in the `homelab` namespace.

Transmission is configured with the `LOCAL_NETWORK` environmental variable: `192.168.1.0/24,10.0.0.0/8`, Pods get their IP in `10.80.0.0/16`, so that matches.

I'm unsure what is up here, or if this is even the place to ask. But some advice would be greatly appreciated!

**Comment by TobiasDeBruijn** [comment, marked as answer]
After a lot of digging, I've found the solution.
TL;DR: Add your service CIDR to the LOCAL_NETWORK variable

It worked with the ingress controller, because it proxies the traffic to Transmission, from a pod, so it is on the pod network. However when accessing it via a service, it is on the service CIDR, and thus not on the pod CIDR.

**Reply by martimors** [reply]
Sorry for necroposting here, but I've been digging for quite some time.

Did you also manage to connect the other way, eg. from transmission and out to the cluster?
